### PR TITLE
Feature/formsubmit

### PR DESCRIPTION
**Pull request by falconmick**
I had to re-write the form submit to add the fix that removes wp credentials from the client side source control..

you will also see that I now am using the following config structure:

app/config/app.js
server/config/app.js
env.js

this is so that anybody writing client side code should be able to tell that they are NOT to include the server config, because if they do they pull in the WP_AUTH which is bad mkay.

env.js is now on the root as having in its own config folder felt dumb, but i didn't want it in server config as it's used to start the server and isn't actually a part of the server.

### CODE REVIEW

File: app/config/app-template.js [hunk 1]
```diff
@@ -1,10 +1,9 @@
-import { HOST, PORT, ENV } from './env';
+import { HOST, PORT, ENV } from '../../env';
 
 // Wordpress
 export const SITE_NAME = 'Starward';
-export const WP_URL = 'http://localhost/starward_wp';
-export const WP_API = `${WP_URL}/wp-json`;
-export const WP_AUTH = new Buffer('username:password').toString('base64');
+export const WP_URL = 'http://localhost/starward_wp'; // put on server side once apollo done
```
Comment (falconmick) [inline app/config/app-template.js]: originally this comment was added as I was still using the wp url for the form, I no longer am, however a utility function for the Navigation that finds out if this is the active link. We need to fix this in issue #76

**Comment by falconmick** [comment]
Please note: this changes how we submit forms. Before we submitted direct to Wordpress, now we are proxying through starward. In the process of doing so I noticed that we were not handling error returns so I have fixed that and now when we recieve errors we work still!

File: app/components/GravityForm/Form.jsx [hunk 2]
```diff
@@ -51,6 +51,8 @@ class GravityForm extends Component {
     const { title, description, button, fields, confirmation } = activeForm;
     // Submit failed watcher
     const { submitFailed } = this.state;
+    const currentForm = this.props.gravityforms[this.props.formId];
+    const postFailed = currentForm && currentForm.hasSubmitted && !currentForm.submitting && !currentForm.submitSuccess;
```
Comment (falconmick) [inline app/components/GravityForm/Form.jsx]: f we have a current form and it has submitted, we then ensure that we are not submitting as a failed submittion previously that is currently being re-submitted is ok, then if we have a form, its submitted atleast once it's not longer submitting AND it wasn't successful, then post failed

**Comment by samlogan** [comment]
Sorry @falconmick, one thing... Can you update the readme before merging?

**Comment by falconmick** [comment]
All good now @samlogan ?

**Comment by samlogan** [comment]
🏆 merge away!

**Pull request merged**